cub/benchmarks/bench/segmented_reduce/base.cuh

```diff
@@ -0,0 +1,154 @@
+                             items_per_thread,
+                             AccumT,
+                             items_per_vec_load,
+                             cub::BLOCK_REDUCE_WARP_REDUCTIONS,
+                             cub::LOAD_LDG>;
+  };
+
+  using MaxPolicy = policy_t;
+};
+#endif // !TUNE_BASE
+
+template <typename T, typename OffsetT>
+void fixed_size_segmented_reduce(nvbench::state& state, nvbench::type_list<T, OffsetT>)
+{
+  using accum_t        = T;
+  using input_it_t     = const T*;
+  using output_it_t    = T*;
+  using offset_t       = cub::detail::choose_offset_t<OffsetT>;
+  using segment_size_t = offset_t;
+  using output_t       = T;
+  using init_t         = T;
+
+#if !TUNE_BASE
+  using policy_t   = policy_hub_t<accum_t>;
+  using dispatch_t = cub::detail::reduce::
+    DispatchFixedSizeSegmentedReduce<input_it_t, output_it_t, offset_t, segment_size_t, op_t, init_t, accum_t, policy_t>;
+#else // TUNE_BASE
+  using dispatch_t = cub::detail::reduce::
+    DispatchFixedSizeSegmentedReduce<input_it_t, output_it_t, offset_t, segment_size_t, op_t, init_t, accum_t>;
+#endif // TUNE_BASE
+
+  // Retrieve axis parameters
+  const size_t num_segments = static_cast<size_t>(state.get_int64("NumSegments{io}"));
+  const size_t segment_size = static_cast<size_t>(state.get_int64("SegmentSize{io}"));
+  const size_t elements     = num_segments * segment_size;
+
+  const size_t max_elements = std::pow(2, 30);
+  if (elements > max_elements)
+  {
+    state.skip("Input size too large for current configuration");
+    return;
+  }
+
+  thrust::device_vector<T> in = generate(elements);
+  thrust::device_vector<T> out(num_segments);
+
+  input_it_t d_in   = thrust::raw_pointer_cast(in.data());
+  output_it_t d_out = thrust::raw_pointer_cast(out.data());
+
+  // Enable throughput calculations and add "Size" column to results.
+  state.add_element_count(elements);
+  state.add_global_memory_reads<T>(elements, "Size");
+  state.add_global_memory_writes<T>(num_segments);
+
+  // Allocate temporary storage:
+  std::size_t temp_size;
+  dispatch_t::Dispatch(
+    nullptr,
+    temp_size,
+    d_in,
+    d_out,
+    num_segments,
+    static_cast<segment_size_t>(segment_size),
+    op_t{},
+    init_t{},
+    0 /* stream */);
+
+  thrust::device_vector<nvbench::uint8_t> temp(temp_size);
+  auto* temp_storage = thrust::raw_pointer_cast(temp.data());
+
+  state.exec(nvbench::exec_tag::no_batch, [&](nvbench::launch& launch) {
+    dispatch_t::Dispatch(
+      temp_storage,
+      temp_size,
+      d_in,
+      d_out,
+      num_segments,
+      static_cast<segment_size_t>(segment_size),
+      op_t{},
+      init_t{},
+      launch.get_stream());
+  });
+}
+
+NVBENCH_BENCH_TYPES(fixed_size_segmented_reduce, NVBENCH_TYPE_AXES(value_types, offset_types))
+  .set_name("base")
+  .set_type_axes_names({"T{ct}", "OffsetT{ct}"})
+  .add_int64_power_of_two_axis("NumSegments{io}", nvbench::range(18, 28, 2))
```

Comment:
suggestion: from the data we saw, it looks like we can reduce the number of values on this axis

Reply:
Not sure if this relates to [Georgii's comment here](https://github.com/NVIDIA/cccl/pull/3969/files#r1980521807), but my intuition would have been to make the number of `Elements` an axis and then implicitly compute `NumSegments` from `Elements/SegmentSize`?